core/vm/evm.go

```diff
@@ -412,6 +421,21 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 	if !evm.Context.CanTransfer(evm.StateDB, caller.Address(), value) {
 		return nil, common.Address{}, gas, ErrInsufficientBalance
 	}
+	// For creation transaction (evm.depth == 0) EIP-3860 checks are done when calculating intrinsic gas
+	if evm.Config.HasEip3860() && evm.depth != 0 {
```

Comment:
A bit hacky way to check if we're in CREATE/CREATE2 and not in creation transaction.